### PR TITLE
Feat/297910  owner duplicate

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

File: app/repos/people.js [hunk 1]
```diff
@@ -70,46 +101,75 @@ const createPeople = async (owners, transaction) => {
...
-      order: [[sequelize.col('person.addresses.address.id'), 'DESC']],
-      include: [{
-        model: sequelize.models.person,
-        where: { person_reference: reference },
-        as: 'person',
-        include: [{
+    const person = await sequelize.models.person.findAll({
+      order: [[sequelize.col('addresses.address.id'), 'DESC']],
+      where: { person_reference: reference },
+      // as: 'person',
```
Comment (jbarnsley10) [inline app/repos/people.js:132]: Can we remove the commented line please?

File: app/repos/mappers/person.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @param {PersonDao} personDao
+ * @returns {CreatedPersonDao}
+ */
+const mapPersonDaoToCreatedPersonDao = (personDao) => {
+  const [personDaoAddressParent] = personDao.addresses
```
[the remark this answers is not in the document]
Comment (whitewaterdesign) [reply app/repos/mappers/person.js:6]: cool, didn't know that that was the purpose of it - thanks, was struggling with finding a good name for that one.

File: test/db-helper.js [hunk 3]
```diff
@@ -1,8 +1,21 @@
 require('../app/data')
 const sequelize = require('../app/config/db')
+const { getCountry } = require('../app/lookups')
 
 const truncate = async () => {
```
Comment (jbarnsley10) [inline test/db-helper.js:5]: Suggest changing method name to 'truncateDatabase' or 'cleanDatabase' ?? 'truncate' suggests it operates on a single table perhaps. Only a suggestion though

File: app/routes/cdo.js [hunk 4]
```diff
@@ -35,11 +36,17 @@ module.exports = [{
...
+        const created = await createCdo(request.payload, getCallingUser(request))
+        const res = cdoCreateDto(created)
 
-      const res = cdoCreateDto(created)
-
-      return h.response(res).code(200)
+        return h.response(res).code(200)
+      } catch (e) {
+        if (e instanceof NotFoundError) {
+          return h.response().code(422).takeover()
```
[the remark this answers is not in the document]
Comment (whitewaterdesign) [reply app/routes/cdo.js:46]: as per conversation:

422 response status code indicates
  that the server understands the content type of the request entity, and the syntax of
  the request entity is correct, but it was unable to process the contained instructions.

File: test/db-helper.js (hunk 3 above)
Comment (whitewaterdesign) [reply test/db-helper.js:5]: Got this one from the original Defra example, but agree ``truncateDatabase`` is clearer.

File: test/mocks/person.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  last_name: 'Skywalker',
+  person_reference: 'P-6076-A37C',
+  birth_date: '1951-09-25',
+  address: {
+    id: 1,
+    address_line_1: 'Moisture Farm',
+    address_line_2: null,
+    country: {
+      id: 22,
+      country: 'Tatooine'
```
Comment (jbarnsley10) [inline test/mocks/person.js:48]: Nice reference :)

**Review by jbarnsley10 — APPROVED** [review]
Great work. A couple of minor comments

**Pull request merged**